api/src/pcapi/routes/pro/users.py

```diff
@@ -300,8 +302,16 @@ def anonymize() -> None:
         raise ResourceNotFoundError(errors={"global": "Cette fonctionnalité n'est pas disponible"})
 
     user = current_user._get_current_object()
+    user_email = user.email
     anonymized = anonymize_pro_user(user)
-    if not anonymized:
+    if anonymized:
+        on_commit(
+            partial(
+                transactional_mails.send_anonymization_confirmation_email_to_pro,
+                user_email,
+            )
+        )
```

Comment:
`on_commit` est désormais utilisé dans `mails.send()`, tu n'en as donc pas besoin ici.
```suggestion
        transactional_mails.send_anonymization_confirmation_email_to_pro(user_email)
```